### PR TITLE
Restore null check

### PR DESCRIPTION
**Pull request by ryanbrandenburg**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ public MediaType(string mediaType, int offset, int? length)
                 throw new ArgumentOutOfRangeException(nameof(offset));
             }
 
-            if (length < 0 || length > mediaType.Length)
+            if (length != null && (length < 0 || length > mediaType.Length))
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs]: Does the null check need to also apply to the next `if`?

Comment (ryanbrandenburg) [inline src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs]: Yes, I :up::date:'ed it.

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs]: Could we just merge the two conditions? 

`if (length != null)
{
  ....
}
```

Might be easier to read

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Formatters/MediaTypeTest.cs [hunk 2]
```diff
@@ -61,6 +61,16 @@ public void Constructor_CanParseMediaTypesWithParameters(string mediaType)
         }
 
         [Fact]
+        public void Constructor_NullLength_IgnoresLength()
+        {
+            // Arrange & Act
+            var result = new MediaType("mediaType", 1, null);
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Formatters/MediaTypeTest.cs]: `length: null`

**Pull request merged**